internal/provider/alert_resource_test.go

```diff
@@ -380,18 +380,23 @@ func Test_ValidateConditions_LengthLessThanOne(t *testing.T) {
 	}
 	diagnosticError := model.validateConditions()
 
-	if len(diagnosticError) != 1 {
-		t.Errorf("expected only one diagnosticError")
-	}
-
-	if diagnosticError[0].attributeName != "conditions" {
-		t.Errorf("unexpected diagnosticError[0].attributeName")
+	expected := []diagnosticsError{
+		{
+			attributeName: "conditions",
+			summary:       "Invalid number of alerting conditions.",
+			details:       "Number of alerting conditions must be between 1 and 5.",
+		},
 	}
-	if diagnosticError[0].summary != "Invalid number of alerting conditions." {
-		t.Errorf("unexpected diagnosticError[0].summary")
+	if len(diagnosticError) != len(expected) {
+		t.Fatalf("expected %v diagnosticErrors", len(expected))
 	}
-	if diagnosticError[0].details != "Number of alerting conditions must be between 1 and 5." {
-		t.Errorf("unexpected diagnosticError[0].details")
+
+	for i := 0; i < len(expected); i++ {
+		if diagnosticError[i] != expected[i] {
+			t.Fatalf("expected(%v, %v, %v) unexpected(%v, %v, %v) ",
+				expected[i].attributeName, expected[i].summary, expected[i].details,
+				diagnosticError[i].attributeName, diagnosticError[i].summary, diagnosticError[i].details)
+		}
 	}
```

Comment:
You could use `reflect.DeepEqual` here to check for equality between the actual and expected arrays, rather than using a loop to check one by one. Also, FYI: this kind of for loop is seldom used in Go. A `range` is preferred when possible. You could do this like `for i := range expected { ... }`. But a `DeepEqual` check is equally effective and more concise, and will give you all the differences at once. Note that in this case, the `Fatalf` call means that you're stopping after the first difference, potentially obscuring differences in subsequent elements as well.

Reply:
I wrote this comment about `DeepEqual` before looking at the `terraform-plugin-framework` package. I left it here for completeness, but I really suggest you go with the package-provided types.